packages/braid-design-system/src/lib/components/Disclosure/Disclosure.tsx

```diff
@@ -40,6 +47,10 @@ export const Disclosure = ({
     "'collapseLabel' must be a string",
   );
 
+  const textContext = useContext(TextContext);
+  const headingContext = useContext(HeadingContext);
+  const isInline = textContext || headingContext;
```

Comment:
Just to be extra safe, since `textContext` can be an object:

```suggestion
  const isInline = Boolean(textContext || headingContext);
```